ReleaseProcessAutomation.IntegrationTests/Steps/JiraSubStepTests.cs

```diff
@@ -0,0 +1,192 @@
+public class JiraSubStepTests
+{
+  private const string c_jiraUrl = "https://re-motion.atlassian.net/";
+  private const string c_jiraProjectKey = "SRCBLDTEST";
+
+  private Mock<IJiraRestClientProvider> _restClientProviderMock;
+  private Mock<IInputReader> _inputReaderMock;
+  private TestConsole _console;
+  private JiraRestClient _restClient;
+  private JiraProjectVersionService _service;
+  private JiraProjectVersionRepairer _repairer;
+  private JiraProjectVersionFinder _versionFinder;
+  private JiraIssueService _issueService;
+  private JiraVersionCreator _jiraVersionCreator;
+  private JiraVersionReleaser _jiraVersionReleaser;
+  private Config _config;
+  private SemanticVersionParser _parser = new SemanticVersionParser();
+
+  [SetUp]
+  public void SetUp ()
+  {
+    _inputReaderMock = new Mock<IInputReader>();
+    _console = new TestConsole();
+    var testCredentials = JiraTestUtility.GetLocallySavedCredentials();
+
+    _restClient = JiraRestClient.CreateWithBasicAuthentication(c_jiraUrl, testCredentials);
+
+    _restClientProviderMock = new Mock<IJiraRestClientProvider>();
+    _restClientProviderMock.Setup(_ => _.GetJiraRestClient()).Returns(_restClient);
+
+    _versionFinder = new JiraProjectVersionFinder(_restClientProviderMock.Object);
+    _issueService = new JiraIssueService(_restClientProviderMock.Object);
+    _service = new JiraProjectVersionService(_restClientProviderMock.Object, _issueService, _versionFinder);
+    _repairer = new JiraProjectVersionRepairer(_service, _versionFinder);
+    _config = new Config
+              {
+                  Jira = new JiraConfig
+                         {
+                             JiraProjectKey = "SRCBLDTEST"
+                         }
+              };
+    _jiraVersionCreator = new JiraVersionCreator(_config, _versionFinder, _service);
+    _jiraVersionReleaser = new JiraVersionReleaser(_config, _repairer, _service);
+  }
+
+  [Test]
+  public void TestFullJiraRun_WithIssueMoving_ShouldMoveProperIssues ()
+  {
+    JiraTestUtility.DeleteVersionsIfExistent(c_jiraProjectKey, _restClient, "4.1.0", "4.1.1", "4.1.2", "4.2.0");
+
+    _inputReaderMock.Setup(_ => _.ReadConfirmation(It.IsAny<bool>())).Returns(true);
+    _service = new JiraProjectVersionService(_restClientProviderMock.Object, _issueService, _versionFinder);
+
+    // Create versions
+    _service.CreateVersion(c_jiraProjectKey, "4.1.0", DateTime.Today.AddDays(1));
+    _service.CreateSubsequentVersion(c_jiraProjectKey, "4\\.1\\..*", 3, DayOfWeek.Monday);
+    _service.CreateSubsequentVersion(c_jiraProjectKey, "4\\.1\\..*", 3, DayOfWeek.Tuesday);
+    _service.CreateVersion(c_jiraProjectKey, "4.2.0", DateTime.Today.AddDays(7));
+
+    // Get latest unreleased version
+    var versions = _versionFinder.FindUnreleasedVersions(c_jiraProjectKey, "4.1.").ToList();
+
+    var versionToRelease = versions.First();
+
+    var additionalVersion = _versionFinder.FindUnreleasedVersions(c_jiraProjectKey, "4.2.").First();
```

Comment:
Here we could also consider using `.Single` instead of `.First`. The difference is that `.Single` throws if there are more than one element in enumerable. In this case it would add the implicit assertion that we expect only one version with the "4.2." prefix.

I will stop commenting on the issue but there are a few more cases of this in this file